nimare/dataset/base.py

```diff
@@ -1,64 +1,84 @@
 # -*- coding: utf-8 -*-
 # emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
 # vi: set ft=python sts=4 ts=4 sw=4 et:
-import json
+"""
+Base classes for datasets.
+"""
+from ..base import ConnMatrix, Image, ActivationSet, Surface
+from nimare.utils import listify
 
 
-class Analyzable(object):
-    def to_array(self):
-        pass
-
+class Study(object):
+    ''' Represents a single published or unpublished study--including one or
+    more contrasts, and study-level metadata. '''
 
-class Mappable(Analyzable):
-    def to_vol(self):
-        pass
+    def __init__(self, contrasts=None, **kwargs):
+        self.contrasts = contrasts or []
 
+    def add_contrasts(self, contrasts):
+        self.contrasts.extend(listify(contrasts))
 
-class ConnMatrix(Analyzable):
-    """Container for connectome data (i.e., connectivity matrices).
-    """
-    def __init__(self, mat):
-        pass
-
-    def to_array(self):
+    @classmethod
+    def merge(cls, studies):
+        ''' Harmonizes/merges Contrasts extracted from different sources, based
+        on common indexes (e.g., DOIs / table numbers, etc.).
+        '''
         pass
 
 
-class Image(Mappable):
-    """Container for volumetric brain images.
+class Contrast(object):
+    """Container for contrasts (aka experiments or comparisons) nested within
+    studies.
+    Should store an arbitrary number of ConnMatrices, Images, and Surfaces,
+    along with at most one ActivationSet.
     """
-    def __init__(self, nimg):
-        pass
-
-    def to_array(self, masker):
-        pass
-
-    def to_vol(self):
-        pass
-
-
-def ActivationSet(Mappable):
-    """Container for peak information, with optional additional metadata (e.g.,
-    intensity values).
-    """
-    def __init__(self, foci):
-        pass
-
-    def to_array(self, method, masker):
-        pass
-
-    def to_vol(self, method, masker):
-        pass
-
-
-def Surface(Mappable):
-    """Container for surface brain data (i.e., from gifti files).
-    """
-    def __init__(self, gimg):
-        pass
-
-    def to_array(self, masker):
-        pass
 
-    def to_vol(self, masker):
+    def __init__(self, images=None, conn_matrices=None, activations=None,
+                 surfaces=None):
+        # Add validation method instead of doing this here. That way it can be
+        # applied to other objects. Should check things like name, min/max,
+
+        self.images = []
+        self.activations = []
+        self.connectomes = []
+        self.surfaces = []
+
+        if images:
+            self.add_images(images)
+
+        if not isinstance(activations, ActivationSet) and activations is not None:
+            raise ValueError('Input activations must be ActivationSet or None')
+        else:
+            self.coordinates = activations
```

Comment:
Addressing a concern raised in #2, I switched to using `coordinates` and `CoordinateSet` in my fork for consistency.